File: Data Structures & Algorithms/products-of-array-discluding-self/submission-5.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n = nums.size();
        vector<int> left(n) , right(n) , ans(n , 1);
        int temp1 = 1, temp2 = 1;
        for(int i =0 ; i< n ;i++){
            temp1 *= nums[i];
            temp2 *= nums[n - i - 1];
            cout << temp1 << " " << temp2 << endl;
            right[i] = temp1;
            left[n - i - 1] = temp2;
        }
        for(int i =0 ; i< n ;i++){
            cout << "before ans :" << ans[i] << endl;
            int l = i -1 , r = i + 1;
            if(l >= 0){
                cout << " yes left !" << right[l];
                ans[i] *= right[l];
            }
            if(r < n){
                cout << " yes right !" <<left[r];
                ans[i] *= left[r];
            }
            cout << "after ans :" << ans[i] << endl;
        }

        return ans;
    }
};

File: Data Structures & Algorithms/find-first-and-last-position-of-element-in-sorted-array/submission-5.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        if(nums.length == 1 && nums[0] == target){
            int ans[] = {0,0};
            return ans;
        }
        else if(nums.length < 1){
            int ans[] = {-1,-1};
            return ans;
        }
        int l = 0 , r = nums.length -1, m = nums.length / 2;
        while(l < r){
            if(nums[m] < target){
                l = m + 1;
            }
            else if(nums[m] > target){
                r = m -1;
            }
            else{
                break;
            }
            m = (r - l) / 2 + l;
        }
        if(nums[m] != target){
            int ans[] = {-1,-1};
            return ans;
        }
        l = m; 
        r = m;
        while(l > 0 && nums[l-1] == target) l--; // Expand left
        while(r < nums.length - 1 && nums[r+1] == target) r++; // Expand right
        return new int[]{l, r};
    }
}